sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricName.java

```diff
@@ -0,0 +1,33 @@
+package org.apache.beam.sdk.metrics;
+
+import com.google.auto.value.AutoValue;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
+
+/**
+ * The name of a metric.
+ */
+@Experimental(Kind.METRICS)
+@AutoValue
+public abstract class MetricName {
+
+  /** The namespace that declared this metric. */
+  public abstract String getNamespace();
+
+  /**
+   * The name of this metric. Should only be null in {@link MetricFilter MetricFilters} being used
```

Comment:
One pattern for search queries is to specify each dimension filter separately. Each dimension can define its own semantics, and the overall filter takes the intersection of all dimensions.

In this case, you might have a `MetricNameFilter` class with `is(MetricName)` and `hasNamespace(String)` factory methods. And a `MetricsQuery` which takes filters on different dimensions.